R/groupingsets.R

```diff
@@ -57,6 +57,14 @@ groupingsets.data.table = function(x, j, by, sets, .SDcols, id = FALSE, jj, ...)
     stopf("Argument 'sets' must be a list of character vectors.")
   if (!is.logical(id))
     stopf("Argument 'id' must be a logical scalar.")
+  if (!(is.null(label) ||
+        (is.atomic(label) && length(label) == 1L) ||
+        (is.list(label) && all(vapply_1b(label, is.atomic)) && all(lengths(label) == 1L) && !is.null(names(label)))))
+    stopf("Argument 'label', if not NULL, must be a scalar or a named list of scalars.")
+  if (is.list(label) && !is.null(names(label)) && ("" %chin% names(label) || anyNA(names(label))))
```

Comment:
is this `!is.null(names(label))` check redundant? Since we have `is.list(label) && ... !is.null(names(label))` in the above requirement?

Reply:
I do find it a bit surprising that the above check requires "a named list of scalars" but we have a separate test for "all list elements must be named", maybe best to add in the check for `""`/`NA` names to the above condition?

Reply:
We can address this as a small follow-up PR if you agree, don't want to hold the PR back further.

Reply:
> is this `!is.null(names(label))` check redundant?

Yes, I think you're right.

> I do find it a bit surprising that the above check requires "a named list of scalars" but we have a separate test for "all list elements must be named", maybe best to add in the check for `""`/`NA` names to the above condition?

Maybe, but with separate checks the error messages can be more specific, the second one being for the situation where `label` is a named list but not all elements have a name. If we combine the error messages into one, it would be something like "Argument 'label', if not NULL, must be (1) a scalar, or (2) a named list with each element being named and each element being a scalar." Or "Argument 'label', if not NULL, must be (1) a scalar, or (2) a named list with no names being "" or NA and each element being a scalar." I think these are less clear than having separate error messages depending on the situation.